astropy/table/operations.py

```diff
@@ -951,7 +951,9 @@ def get_descrs(arrays, col_name_map):
             raise TableMergeError(f'Key columns {names!r} have different shape')
         shape = uniq_shapes.pop()
 
-        out_descrs.append((fix_column_name(out_name), dtype, shape))
+        if out_name is not None:
```

Comment:
I see you use in-line `if` to do such conversion, why not make that style consistent throughout this PR?

Reply:
I used the one-liner if the variable on the left hand side had a different name than the variable on the right hand side, but if you think it's worth it then I could use it here too.

Reply:
Naw, it's ok. Was just a nitpick.